app/src/test/java/com/codesoom/assignment/controllers/TaskControllerNestedTest.java

```diff
@@ -0,0 +1,260 @@
+package com.codesoom.assignment.controllers;
+
+import com.codesoom.assignment.TaskNotFoundException;
+import com.codesoom.assignment.application.TaskService;
+import com.codesoom.assignment.models.Task;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+class TaskControllerNestedTest {
+
+    private final String DEFAULT_TITLE = "TEST";
+    private final long DEFAULT_SIZE = 3;
+    private TaskController controller;
+
+    @BeforeEach
+    void setUp() {
+        // subject
+        controller = new TaskController(new TaskService());
+
+        // fixtures
+    }
+
+    void setDefaultSizeTasks(){
+        for(long i = 1 ; i <= DEFAULT_SIZE ; i++){
+            Task newTask = new Task();
+            newTask.setTitle(DEFAULT_TITLE + i);
+            controller.create(newTask);
+        }
+    }
+
+    @Nested
+    @DisplayName("list 메소드는")
+    class Describe_List{
+
+        @Nested
+        @DisplayName("Task가 존재한다면")
+        class Context_NotEmptyTask{
+
+            @Test
+            @DisplayName("모든 Task를 반환한다")
+            void It_ReturnAllTask(){
+                setDefaultSizeTasks();
+                assertThat(controller.list()).hasSize((int) DEFAULT_SIZE);
+                for(long i = 1 ; i <= DEFAULT_SIZE ; i++){
+                    assertThat(controller.detail(i)).isEqualTo(new Task(i , DEFAULT_TITLE + i));
+                }
+            }
+        }
+
+        @Nested
+        @DisplayName("Task가 존재하지 않는다면")
+        class Context_EmptyTask{
+
+            @Test
+            @DisplayName("비어있는 List를 반환한다")
+            void It_ReturnEmptyList(){
+                assertThat(controller.list()).hasSize(0);
+            }
+        }
+
+    }
+
+    @Nested
+    @DisplayName("detail 메소드는")
+    class Describe_Detail{
+
+        @Nested
+        @DisplayName("id가 null이라면")
+        class Context_NullPathVariable{
+
+            @Test
+            @DisplayName("TaskNotFoundException이 발생한다")
+            void It_ThrowException(){
+                assertThatThrownBy(() -> controller.detail(null))
+                        .isInstanceOf(TaskNotFoundException.class);
+            }
+        }
```

Comment:
`id`가 `null`인 경우에 대한 Context를 만들었으므로, Context 영역에서 아이디를 선언해서 사용해 보세요. 예를 들자면 이렇게요.


```suggestion
        @Nested
        @DisplayName("id가 null이라면")
        class Context_NullPathVariable{
            private final givenId = null;
            @Test
            @DisplayName("TaskNotFoundException를 던진다")
            void It_ThrowException(){
                assertThatThrownBy(() -> controller.detail(givenId))
                        .isInstanceOf(TaskNotFoundException.class);
            }
        }
```

한편 `~예외가 발생한다`라는 표현에 대해 생각해볼 점이 하나 있습니다. 그것은 우리가 예외를 취급하는 코드를 작성할 때 `throw`, `catch`, `@throws`... 같은 단어를 쓴다는 거죠. 79번 라인의 assert 메소드 이름을 읽어보세요. `assertThatThrownBy`라고 되어 있죠?

우리가 애플리케이션의 사용자라면 예외가 언제 던져질지 모릅니다. 그래서 예외는 "발생하는" 것이라 생각하게 되죠. 그런데 우리는 애플리케이션을 만들고 있습니다. 언제 어떤 예외를 던질지 결정하는 건 우리죠. 그래서 이렇게 설명을 적게 됩니다.

- `detail 메소드는, id 가 null 이라면 TaskNotFoundException 예외를 던진다.`

Reply:
> 우리가 애플리케이션의 사용자라면 예외가 언제 던져질지 모릅니다. 그래서 예외는 "발생하는" 것이라 생각하게 되죠.
우리는 애플리케이션을 만들고 있습니다. 

말씀대로 애플리케이션을 만드는 사람이 던져주는 것인데 "발생한다"라고 무의식적으로 생각하게 되는 것 같습니다 ㅎㅎ
명심하겠습니다